development/src/components/welcome_comp.js

```diff
@@ -6,44 +6,50 @@ import SignUpComponent from "./signup_comp";
 import { usePathname, useRouter, useSearchParams } from "next/navigation";
 import { useEffect } from "react";
 import UserLoginComp from "./userLogin_comp";
+import VerificationOverlay from "./VerificationOverlay";
 import { ForgotPassword } from "./ForgotPassword";
 
 const Welcome = () => {
   const router = useRouter();
   const view = useSearchParams().get("view");
   return (
     <div>
-      {view == "login" ? (
-        <Modal
-          onClose={() => {
-            router.push("/");
-          }}
-        >
-          <UserLoginComp
-            onClose={() => {
-              router.push("/");
-            }}
-          />
-        </Modal>
-      ) : view == "signup" ? (
-        <Modal
-          onClose={() => {
-            router.push("/");
-          }}
-        >
-          <SignUpComponent />
-        </Modal>
-      ) : view == "recoveraccount" ? (
-        <Modal
-          onClose={() => {
-            router.push("/");
-          }}
-        >
-          <ForgotPassword />
-        </Modal>
-      ) : (
-        <div></div>
-      )}
+    {view === "login" ? (
+      <Modal
```

Comment:
This shouldn’t be on welcome page, it should be applied in `page.js` as the verification modal won’t be visible if welcome page is closed when you try to route to the `verificationOverlay`.